### PR TITLE
Storage/feature/stg87-hns encryption context

### PR DESCRIPTION
**Pull request by microzchang**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

File: sdk/storage/azure-storage-files-datalake/test/ut/datalake_file_client_test.cpp [hunk 1]
```diff
@@ -469,6 +469,34 @@ namespace Azure { namespace Storage { namespace Test {
     }
   }
 
+  TEST_F(DataLakeFileClientTest, DISABLED_CreateWithEncryptionContext)
+  {
+    std::string encryptionContext = "encryptionContext";
+    std::string fileName = GetTestNameLowerCase() + "_file";
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/test/ut/datalake_file_client_test.cpp:475]:
```suggestion
    const std::string fileName = RandomString();
```

**Review by Jinming-Hu — COMMENTED** [review]
The new field `EncryptionContext` is non-nullable in both response and request options in .Net SDK. Can you check other SDK as well?

**Comment by microzchang** [comment]
> The new field `EncryptionContext` is non-nullable in both response and request options in .Net SDK. Can you check other SDK as well?

I checked Java SDK, it's not a required value(which is nullable in java).
https://github.com/Azure/azure-sdk-for-java/blob/ac1e8b0c5c89603ea659710af39e5fcf187d26b9/sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/Transforms.java#L330

And I also asked @EmmaZhu , she also thought it's a nullable value.

**Review by Jinming-Hu — APPROVED** [review]

**Pull request merged**